packages/shared/src/hooks/usePostModalNavigation.ts

```diff
@@ -86,12 +88,21 @@ export const usePostModalNavigation = (
     scrollPositionOnFeed.current = 0;
   };
 
+  useEffect(() => {
+    router.events.on('routeChangeStart', onCloseModal);
```

Comment:
This might close the modal when you navigate through its header (left and right arrows). Can you check if it does not?

Reply:
Probably testable on https://hotfix-close-post-modal-nav.preview.app.daily.dev/ 😉

Reply:
Yeah, we have to check if it is in `/posts/` right?

Reply:
@sshanzel valid concern, see here seems to be working as expected:

https://github.com/dailydotdev/apps/assets/554874/ef4a7438-5027-47ae-b682-d0e445383549

Reply:
Yeah, looks to be working due to using shallow history routing for back/forward arrows. Looks ok from my end 👍

Reply:
Ah yes, let's add that to be safe, see small edge-case on first click

Reply:
@sshanzel // @capJavert added the failsafe anyway :)